deployment/terraform/agent.go

```diff
@@ -44,41 +46,84 @@ func (t *Terraform) generateLoadtestAgentConfig(output *terraformOutput) loadtes
-	cmd := "cd mattermost-load-test-ng && export PATH=$PATH:/usr/local/go/bin && go run ./cmd/ltagent init"
-	if out, err := sshc.RunCommand(cmd); err != nil {
-		return fmt.Errorf("error running ssh command: %s, out: %s, error: %w", cmd, out, err)
+	for _, val := range output.Agents.Value {
+		sshc, err := extAgent.NewClient(val.PublicIP)
+		if err != nil {
+			return err
+		}
+		mlog.Info("configuring agent", mlog.String("ip", val.PublicIP))
+		if uploadBinary {
+			// upload the local file with a -local prefix to avoid collision
+			// with the previously downloaded file
+			dstFile := "/home/ubuntu/mattermost-load-test-ng-local.tar.gz"
+			mlog.Info(fmt.Sprintf("Uploading binary file %s", packagePath))
+			if out, err := sshc.UploadFile(packagePath, dstFile, false); err != nil {
+				return fmt.Errorf("error uploading file %q, output: %q: %w", packagePath, string(out), err)
+			}
+			if out, err := sshc.RunCommand("rm -rf mattermost-load-test-ng && " +
+				"tar xzf mattermost-load-test-ng-local.tar.gz && " +
+				"mv $(ls -d */ | grep mattermost-load-test-ng) mattermost-load-test-ng"); err != nil {
+				return fmt.Errorf("error running command, got output: %q: %w", string(out), err)
+			}
+		}
+
+		mlog.Info("Uploading agent service file")
+		rdr := strings.NewReader(strings.TrimSpace(agentServiceFile))
+		if out, err := sshc.Upload(rdr, "/lib/systemd/system/ltagent.service", true); err != nil {
+			return fmt.Errorf("error uploading file, output: %q: %w", string(out), err)
+		}
+
+		mlog.Info("Uploading coordinator service file")
+		rdr = strings.NewReader(strings.TrimSpace(coordinatorServiceFile))
+		if out, err := sshc.Upload(rdr, "/lib/systemd/system/ltcoordinator.service", true); err != nil {
+			return fmt.Errorf("error uploading file, output: %q: %w", string(out), err)
+		}
+
+		mlog.Info("Starting agent")
+		cmd := fmt.Sprintf("sudo service ltagent start")
+		if out, err := sshc.RunCommand(cmd); err != nil {
+			return fmt.Errorf("error running command, got output: %q: %w", string(out), err)
+		}
 	}
 	return nil
 }
 
-func (t *Terraform) configureAndRunAgent(extAgent *ssh.ExtAgent, ip string, output *terraformOutput) error {
+func (t *Terraform) initLoadtest(extAgent *ssh.ExtAgent, ip string, output *terraformOutput) error {
 	sshc, err := extAgent.NewClient(ip)
 	if err != nil {
 		return err
 	}
-
-	loadtestConfig := t.generateLoadtestAgentConfig(output)
-	data, err := json.Marshal(loadtestConfig)
+	mlog.Info("Populating DB", mlog.String("agent", ip))
+	cfg := t.generateLoadtestAgentConfig(output)
+	data, err := json.Marshal(cfg)
 	if err != nil {
 		return err
 	}
 	dstPath := "/home/ubuntu/mattermost-load-test-ng/config/config.json"
+	mlog.Info("Uploading updated config file")
 	if out, err := sshc.Upload(bytes.NewReader(data), dstPath, false); err != nil {
-		return fmt.Errorf("error running ssh command: output: %s, error: %w", out, err)
+		return fmt.Errorf("error uploading file, output: %q: %w", string(out), err)
```

Comment:
Do we really need string(out) here ? I thought fmt would automatically format it to string quoted since it's `%q`